### PR TITLE
Improve tests: multiple Ruby versions, Yabeda RSpec matchers, updates…

### PR DESCRIPTION
**Pull request by Envek**
See individual commits:

 - [Test against multiple versions of Ruby, run Rubocop in separate job](https://github.com/ianks/yabeda-gc/commit/5667a821d87ede7205b60197cf920b2d0fbf48cb)
 - [Use built-in RSpec matchers from Yabeda](https://github.com/ianks/yabeda-gc/commit/045b07a2360f60b4662c06e0bf460e980f3dc57e)
 - [Declare time metric for GC only on Ruby 3.1](https://github.com/ianks/yabeda-gc/commit/4c70c97084682feeab896c98c8d8660c517cbcf9) (follow up for #3)

### CODE REVIEW

**Comment by ianks** [comment]
Thank you!

I missed this PR before I updated some deps, which is causing some merge conflicts. Do you mind fixing those? Also, I invited you as an owner on rubygems.org in case you want to release (using your public github email).

File: .rubocop.yml [hunk 1]
```diff
@@ -18,3 +18,6 @@ Layout/LineLength:
 
 Metrics/BlockLength:
   Enabled: false
+
+Gemspec/RequireMFA:
```
Comment (ianks) [inline .rubocop.yml]: Let's remove this, MFA will soon be required for all releases per rubygems

Comment (Envek) [reply .rubocop.yml]: Done

**Review by ianks — APPROVED** [review]

**Comment by Envek** [comment]
Thanks, released [0.2.1](https://rubygems.org/gems/yabeda-gc/versions/0.2.1)

**Pull request merged**